.github/workflows/check_for_redirects.yml

```diff
@@ -0,0 +1,51 @@
+name: Redirects Workflow
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, labeled]
+env:
+  DIFF_DIRECTORIES: 'src/pages src/fragments'
+jobs:
+  onPrOpen:
+    name: Check if redirects are needed on PR opened
+    runs-on: ubuntu-latest
+    if: github.event.action == 'opened'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+      - name: Get count of deleted files
+        run: |
+          git fetch origin main
+          echo "DELETED_FILES_ON_OPENED=$(git diff --name-status --diff-filter=D origin/main -- ${{ env.DIFF_DIRECTORIES }} ${{ github.pull_request.head.sha }} -- ${{ env.DIFF_DIRECTORIES }} | wc -l)" >> $GITHUB_ENV
+          echo "Deleted file count: ${{ env.DELETED_FILES_ON_OPENED }}"
```

Comment:
As general best practices, 

can we pull `$github.pull_request.head.sha` to its own environment variable, for abundance of caution?

For example: https://github.com/aws-amplify/amplify-swift/blob/main/.github/workflows/cleanup_labels_issue_close.yml#L18-L26